### PR TITLE
docs: Updated CI guide

### LINKED ISSUE #4943: docs: Simplify CI guide
> The python dependencies install could be simplified and the go version I think is 1.20, but that can be in a follow up.

_Originally posted by @helderco in https://github.com/dagger/dagger/pull/4937#pullrequestreview-1380990279_

### PR DESCRIPTION
**Pull request by vikram-dagger**
Closes #4943

### CODE REVIEW

**Comment by vikram-dagger** [comment]
@helderco could you please propose the improvements you mentioned in #4943 for the Python dependencies?

**Comment by helderco** [comment]
It's just:

```diff
- python -m pip install --upgrade pip
- if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
- pip install dagger-io
+ pip install -r requirements.txt
```

For all CIs.

**Comment by helderco** [comment]
Also need to fix last paragraph:
- https://github.com/dagger/dagger/pull/4937#discussion_r1166863983

**Review by helderco — CHANGES_REQUESTED** [review]

**Comment by helderco** [comment]
Just to clarify some reasoning behind https://github.com/dagger/dagger/pull/4952#issuecomment-1508536845:

- While it is good to keep pip up to date, you see that line in some CI examples just to silence a warning that there's a newer pip version. While you can also silence that warning with an env var, I don't think it's worth it to add to the example. It's not required.
- There are multiple tools to manage packages in Python. With any of those the command will be different. If using Poetry, it'd be `poetry install`, for example. It's similar to `npm` vs `yarn` with Node.js but our guide simply uses `npm` because it's built-in. It's the same with these instructions.

TLDR: just keeping the example as simple as possible, as with the other SDKs.

**Review by helderco — APPROVED** [review]
There were a few final touches but I've fixed them.
